### ISSUE TITLE
Export raw k-space data

### ISSUE THREAD

**Issue by bpinsard** [CMRR-C2P/MB#268]
Hi, this is more a question than an issue:
in https://www.ncbi.nlm.nih.gov/pmc/articles/PMC6484842/ it says that the CMRR MB sequence has an:
> option to automatically write the raw k-space data to a remote disk

I was wondering how to enable that. From the doc the only option I see is:
> Online multi-band recon.: Online will reconstruct the multi-band
> accelerated images in ICE and send them to the database. Aliased will
> send the aliased images directly to the database without reconstruction (for
> debugging). None is equivalent to EmptyIce (e.g. for later reconstruction
> offline).
> If the optional remote reconstruction service is configured, the option to
> enable it will be found here.

Is raw k-space export exclusive with online reconstruction? Or does it require the optional remote reconstructor? 

Thanks!

**Comment by eauerbach** [comment]
Yes, that feature uses the remote reconstruction server. I don't know if it is still useful with the latest software versions (maybe Siemens significantly improved their realtime data export tools, I haven't tried lately).

**Comment by mharms** [comment]
@eauerbach : Isn't the "Save reduced raw data" option, which saves the reduced k-space data to a disk installed in the MARS, independent of the old remote recon server?

**Comment by eauerbach** [comment]
That is something else, I don't think it is publicly documented...

**Comment by mharms** [comment]
Is that one of the options that is only exposed if you have appropriate settings in place in the `MBAdvancedSettings.ini` file?  If so, that would explain why it isn't included in the documentation PDF, and why @bpinsard couldn't find it.

**Comment by eauerbach** [comment]
Yes, what you are referring to is on the secret menu, currently only for HCP.

The export feature using the remote recon server was meant to minimize latency for remote (post)processing, specifically for realtime FMRI. It can stream raw or fully reconstructed data. Unfortunately streaming multichannel raw data in real time requires a 10 GbE link, which is not standard. Streaming reconstructed data is generally fine over the standard 1 GbE link. In the past it was considerably faster than using the product host-based "online export" functionality.

**Comment by mharms** [comment]
Partly asking because I just had someone local at WU ask yesterday about saving the raw k-space data for a multi-site study using the CMRR sequences.  I mistakenly thought support for that option was now public, but I see not.  Should I tell him that option is not something that CMRR is supporting, outside of its involvement with HCP proper?

**Comment by bpinsard** [comment]
Thanks for the replies, though I am not sure to fully understand the answers:
- Is it possible to activate that function for CMRR-MB sequence through the `MBAdvancedSettings.ini`
- Is it superseded by more recent Siemens product export options that I am not aware of? If so from which VE11 version is it available? 
- If not used in a real-time application (but to allow post-hoc reconstruction if newer algorithms/bug-fixes are released) is the 1GbE link sufficient?

The interest here, is for longitudinal studies, which would require fully automated export and/or transfer to a dedicated server.
Thanks!

**Comment by whitecraine** [comment]
Can someone advise if requests for remote recon servers are still being accepted? One site I am collaborating with is still on VB17 Trio and is wobdering if this might be an option available to them to speed things up.

**Comment by julfou81** [comment]
Hi, I heard recently about this solution, that allows the automatic raw data export from MARS to an external drive:

https://yarra-framework.org/addons/filemover/

You just have to install it on the syngo console, the installation was quick but I must say that it was done by our Siemens application specialist who had done it before.

You can specify from which sequence you want the raw data to be exported and also schedule when you want it to be exported (beginning of the day, etc).
